src/io/parsers/GROParser.js

```diff
@@ -0,0 +1,230 @@
+   */
+  canProbablyParse() {
+    return _.isString(this._data);
+  }
+
+  /**
+   * Parsing title of molecule complex.
+   * NOTE: that names are ESTIMATES, there is no strict rules in Gromos87 standard for first line in input file.
+   * @param {GROReader} line - Line containing title and time.
+   */
+  _parseTitle(line) {
+    const { metadata } = this._complex;
+    metadata.id = line.readLine().trim();
+    metadata.format = 'gro';
+  }
+
+  /**
+   * Parsing line containing number of atoms information.
+   * @param {GROReader} line - Line containing number of atoms.
+   */
+  _parseNumberOfAtoms(line) {
+    this._numAtoms = line.readInt(0, line.getNext());
+    if (Number.isNaN(this._numAtoms)) {
+      this._complex.error = {
+        message: `"${line.readLine()}" is not representing atom number. Consider checking input file`,
+      };
+    }
+  }
+
+  /**
+   * Parsing line containing information about residues, atoms etc. Also information about box vectors.
+   * Format of atoms MUST (by Gromos87 standard) be this: (note that numbering starts not from 0, but from 1!)
+   * ResidueNumber[1 - 5]  ResidueName[6 - 10] AtomName[11 - 15] AtomNumber[16 - 20] Position[21 - 45] Velocity[46 - 69]
+   * @param {GROReader} line - Line containing information about atom.
+   */
+  _parseAtom(line) {
+    this._residueNumber = line.readInt(1, 5);
+    this._residueName = line.readString(6, 10).trim();
+    this._atomName = line.readString(11, 15).trim();
+    this._atomNumber = line.readInt(16, 20);
+    const positionX = line.readFloat(21, 28) * 10;
+    const positionY = line.readFloat(29, 36) * 10;
+    const positionZ = line.readFloat(37, 45) * 10;
+    if (Number.isNaN(positionX) || Number.isNaN(positionY) || Number.isNaN(positionZ)) {
+      this._complex.error = {
+        message: `Atom position is invalid in "${line.readLine()}"`,
+      };
+      return;
+    }
+    /* const velocityX = line.readFloat(46, 53);
+    const velocityY = line.readFloat(54, 61);
+    const velocityZ = line.readFloat(62, 69); */
+    /* Adding residue and atom to complex structure */
+    const type = Element.getByName(this._atomName[0]); /* MAGIC 0. REASONS: This name is something like "CA", where
+     C - is an element an A is something else. But what about Calcium? */
+    if (type.fullName === 'Unknown') {
+      this._complex.error = {
+        message: `${this._atomName[0]} hasn't been recognised as an atom name.`,
+      };
+      return;
+    }
+    const role = Element.Role[this._atomName];
+    /* Firstly, create a dummy chain */
+    let chain = this._chain;
+    if (!chain) {
+      this._chain = chain = this._complex.addChain('A');
+    }
+    /* Secondly, add residue to that chain */
+    let residue = this._residue;
+    if (!residue || residue.getSequence() !== this._residueNumber) {
+      this._residue = residue = chain.addResidue(this._residueName, this._residueNumber, 'A');
```

Comment:
I'm afraid we shouldn't use 'A' for the insertion code here, it should be `' '` (a space), I believe.